### PR TITLE
Last min fixes

### PR DESCRIPTION
**Pull request by johnEthicalTechnology**
Fixed:
- State would remain the same between use of base filter and base filter + sub filter.
- State wouldn't update when sub filter all records check box is unticked.
- Changes made to filter function caused unexpected results. Removed redundant code and made it clearer.

### CODE REVIEW

File: src/utils/filterResults.ts [hunk 1]
```diff
@@ -66,18 +66,17 @@ export default function filterResults (unsortedPropertyResults: UnprocessedResul
...
-                const allRecordsForSalesEvidenceFilter = searchParams.allRecords
-                canAddBasedOnMaxResults = allRecordsForSalesEvidenceFilter ? true : canAddBasedOnMaxResults && salesEvidenceFilter(searchParams, property)
-            }
+            const canAddBasedOnMaxResults = isUnderNeighbourLimit || isUnderPropertyTypeLimit || isUnderPropertyGroupLimit
 
             if (canAddBasedOnMaxResults) {
                 const propertyTypeMatch = !isPropertyTypeFilterInUse && isUnderPropertyTypeLimit && matchForPropertyTypes(property, desiredPropertyTypes)
                 const propertyGroupMatch = !isPropertyGroupFilterInUse && isUnderPropertyGroupLimit && matchForPropertyGroups(property, desiredPropertyGroups)
-                const canAddBasedOnFilters = propertyGroupMatch || propertyTypeMatch || isUnderNeighbourLimit
+                let canAddBasedOnFilters = propertyGroupMatch || propertyTypeMatch
```
Comment (jeznag) [inline src/utils/filterResults.ts:90]: what about isUnderNeighbourLimit?

File: src/services/crmDataService.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
 import { IntersectedSearchAndFilterParams, UnprocessedResultsFromCRM, PositionType } from '../types'
 import { ZOHO } from '../vendor/ZSDK'
-import emailAndIdExtract from '../utils/emailAndIdExtract'
 import filterResults from '../utils/filterResults'
+import emailAndIdExtract from '../utils/emailAndIdExtract'
```
Comment (jeznag) [inline src/services/crmDataService.ts:4]: sorry I should've run eslint!

File: src/utils/filterResults.ts [hunk 3]
```diff
@@ -67,46 +83,59 @@ export default function filterResults (unsortedPropertyResults: UnprocessedResul
...
+                    if (!isPropertyGroupFilterInUse && !isPropertyTypeFilterInUse) {
+                        canAddBasedOnFilters = true
+                    }
                     // N.B. the Sales Evidence Filter doesn't have the ability to search for multiple properties hence only passing in the single search param object.
-                    const allRecordsForSalesEvidenceFilter = searchParams.allRecords
                     canAddBasedOnFilters = allRecordsForSalesEvidenceFilter ? true : canAddBasedOnFilters && salesEvidenceFilter(searchParams, property)
                 }
 
-                const ownerData = getOwnerData(property)
                 const isManaged = (property.Managed === desiredManaged) || desiredManaged === 'All'
```
Comment (jeznag) [inline src/utils/filterResults.ts:101]: I don't think all these if statements are needed
how about ```

const canAddBasedOnManaged = !isManagedFilterInUse || (property.Managed === desiredManaged) || desiredManaged === 'All'

const shouldAddProperty = canAddBasedOnManaged && canAddBasedOnFilters && isUnderNeighbourLimit
```

Comment (johnEthicalTechnology) [reply src/utils/filterResults.ts:101]: I tried it and it doesn't work on the most basic test of simply selecting "Yes" for Managed and no other filter. So I'm not going to bother testing it for any of the other filter states.

If you want to find another way to get it to work differently to how it's currently working I'm happy for you to work on it. I've create a spreadsheet to show on what filter conditions the current setup passes. I can send you that so as to compare and remain consistent.

Comment (jeznag) [reply src/utils/filterResults.ts:101]: Sorry should be ```const shouldAddProperty = canAddBasedOnManaged || canAddBasedOnFilters || isUnderNeighbourLimit```

share the spreadsheet when you get a chance. Would be good to reduce the cyclomatic complexity of this function to make it easier to understand and maintain

Comment (jeznag) [reply src/utils/filterResults.ts:101]: I decided not to play around with it :P Too busy with other stuff today.

**Pull request merged**